src/realm/sort_descriptor.cpp

```diff
@@ -478,7 +544,14 @@ void BaseDescriptor::Sorter::cache_first_column(IndexPairs& v)
         }
 
         const auto obj = col.table->get_object(key);
-        index.cached_value = ck.get_value(obj);
+        index.cached_string_id = ck.get_string_id(obj);
+        if (index.cached_string_id) {
+            index.cached_value = Mixed();
+        }
+        else {
+            index.cached_value = ck.get_value(obj);
+            // index.cached_string_id = {};
+        }
```

Comment:
Since cached_value and cached_string_id are mutually exclusive, it'd be nice to just store the StringID in cached_value rather than a separate field. Adding 16 bytes per row is not an insignificant increase in the memory required for sorting.